### PR TITLE
Feature/feature adapting libraries

### PR DESCRIPTION
**Pull request by AlbertPG**

### CODE REVIEW

File: app/src/main/java/org/pesmypetcare/mypetcare/services/user/UserManagerAdapter.java [hunk 1]
```diff
@@ -89,69 +101,95 @@ private void assignImageFromServer(User user) throws MyPetCareException {
 
     @Override
     public boolean userExists(User user) {
-        UserData userData = null;
+        AtomicReference<UserData> userData = new AtomicReference<>();;
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/user/UserManagerAdapter.java:104]: Codacy Issue found: [Empty statement.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5635133)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity decreasing per file
==============================
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/user/TrExistsUsername.java  -1
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrUpdateWash.java  -2
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrObtainAllPetWashes.java  -1
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/services/pet/PetManagerAdapter.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/medicalprofile/MedicalProfileManagerAdapter.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/services/googlecalendar/GoogleCalendarAdapter.java  1
         

Clones removed
==============
+ app/src/main/java/org/pesmypetcare/mypetcare/services/vetvisits/VetVisitsAdapter.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5635133&bid=18454556)

File: app/src/main/java/org/pesmypetcare/mypetcare/services/user/UserManagerAdapter.java (hunk 1 above)
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/user/UserManagerAdapter.java:104]: Codacy Issue found: [An empty statement (semicolon) not part of a loop](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5635133)

**Review by xavier-campos-diaz — DISMISSED** [review]

**Review by xavier-campos-diaz — DISMISSED** [review]

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/user/TrExistsUsername.java  -1
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrUpdateWash.java  -2
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrObtainAllPetWashes.java  -1
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/services/pet/PetManagerAdapter.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/googlecalendar/GoogleCalendarAdapter.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/medicalprofile/MedicalProfileManagerAdapter.java  3
         

Clones removed
==============
+ app/src/main/java/org/pesmypetcare/mypetcare/services/vetvisits/VetVisitsAdapter.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5635133&bid=18454556)

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrUpdatePetIllness.java [hunk 2]
```diff
@@ -60,7 +58,7 @@ public void setNewDate(String newDate) {
     /**
      * Executes the transaction.
      */
-    public void execute() throws NotPetOwnerException, ExecutionException, InterruptedException {
```
Comment (marclll) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrUpdatePetIllness.java:63]: Exception not explained

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrUpdatePetVaccination.java [hunk 3]
```diff
@@ -60,7 +58,7 @@ public void setNewDate(String newDate) {
     /**
      * Executes the transaction.
      */
-    public void execute() throws NotPetOwnerException, ExecutionException, InterruptedException {
```
Comment (marclll) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrUpdatePetVaccination.java:63]: exception not explained

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/medication/TrNewPetMedication.java [hunk 4]
```diff
@@ -63,8 +60,7 @@ public boolean isResult() {
     /**
      * Executes the transaction.
      */
-    public void execute() throws MedicationAlreadyExistingException, ExecutionException, InterruptedException,
-        InvalidFormatException {
```
Comment (marclll) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/medication/TrNewPetMedication.java:67]: exception not explained

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/medication/TrUpdateMedication.java [hunk 5]
```diff
@@ -72,7 +70,7 @@ public void setNewName(String newName) {
     /**
      * Execute the transaction.
      */
-    public void execute() throws InterruptedException, ExecutionException, MedicationAlreadyExistingException {
```
Comment (marclll) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/medication/TrUpdateMedication.java:75]: exception not explained

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/user/TrObtainUserImage.java [hunk 6]
```diff
@@ -44,7 +42,7 @@ public Bitmap getResult() {
         return result;
     }
 
-    public void execute() throws ExecutionException, InterruptedException, MyPetCareException {
```
Comment (marclll) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/user/TrObtainUserImage.java:47]: exception not explained

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/vetvisits/TrDeleteVetVisit.java [hunk 7]
```diff
@@ -61,7 +59,7 @@ public boolean isResult() {
     /**
      * Executes the transaction.
      */
-    public void execute() throws NotPetOwnerException, ExecutionException, InterruptedException {
```
Comment (marclll) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/vetvisits/TrDeleteVetVisit.java:64]: exception not explained

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/vetvisits/TrNewVetVisit.java [hunk 8]
```diff
@@ -62,8 +59,7 @@ public boolean isResult() {
     /**
      * Executes the transaction.
      */
-    public void execute() throws VetVisitAlreadyExistingException, NotPetOwnerException, ExecutionException,
-        InterruptedException, InvalidFormatException {
```
Comment (marclll) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/vetvisits/TrNewVetVisit.java:66]: exception not explained

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/vetvisits/TrUpdateVetVisit.java [hunk 9]
```diff
@@ -69,7 +67,7 @@ public boolean isResult() {
     /**
      * Executes the transaction.
      */
-    public void execute() throws NotPetOwnerException, ExecutionException, InterruptedException {
+    public void execute() throws NotPetOwnerException {
```
Comment (marclll) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/vetvisits/TrUpdateVetVisit.java:71]: exception not explained

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrNewPetWash.java [hunk 10]
```diff
@@ -62,8 +59,7 @@ public boolean isResult() {
     /**
      * Execute the transaction.
      */
-    public void execute() throws WashAlreadyExistingException, InterruptedException, ExecutionException,
-        InvalidFormatException {
```
Comment (marclll) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrNewPetWash.java:66]: exception not explained

**Review by marclll — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/exercise/TrAddExercise.java [hunk 11]
```diff
@@ -81,8 +78,7 @@ public void setEndDateTime(DateTime endDateTime) {
     /**
      * Execute the transaction.
      */
-    public void execute() throws NotPetOwnerException, InvalidPeriodException, ExecutionException,
-        InterruptedException, InvalidFormatException {
+    public void execute() throws NotPetOwnerException, InvalidPeriodException {
```
Comment (marclll) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/exercise/TrAddExercise.java:83]: exception not explained

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/exercise/TrDeleteExercise.java [hunk 12]
```diff
@@ -51,7 +49,7 @@ public void setExerciseDateTime(DateTime dateTime) {
     /**
      * Execute the transaction.
      */
-    public void execute() throws NotPetOwnerException, NotExistingExerciseException, ExecutionException, InterruptedException {
+    public void execute() throws NotPetOwnerException, NotExistingExerciseException {
```
Comment (marclll) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/exercise/TrDeleteExercise.java:52]: exception not explained

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/exercise/TrGetAllExercises.java [hunk 13]
```diff
@@ -30,7 +29,7 @@ public void setPet(Pet pet) {
         this.pet = pet;
     }
 
-    public void execute() throws NotPetOwnerException, ExecutionException, InterruptedException {
+    public void execute() throws NotPetOwnerException {
```
Comment (marclll) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/exercise/TrGetAllExercises.java:44]: Function not explained

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/exercise/TrUpdateExercise.java [hunk 14]
```diff
@@ -90,7 +89,7 @@ public void setEndDateTime(DateTime endDateTime) {
     /**
      * Execute the transaction.
      */
-    public void execute() throws NotPetOwnerException, NotExistingExerciseException, InvalidPeriodException, ExecutionException, InterruptedException {
+    public void execute() throws NotPetOwnerException, NotExistingExerciseException, InvalidPeriodException {
```
Comment (marclll) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/exercise/TrUpdateExercise.java:95]: exception not explained

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/meals/TrDeleteMeal.java [hunk 15]
```diff
@@ -51,7 +49,7 @@ public void setMeal(Meals meal) {
     /**
      * Executes the transaction.
      */
-    public void execute() throws NotPetOwnerException, ExecutionException, InterruptedException {
+    public void execute() throws NotPetOwnerException {
```
Comment (marclll) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/meals/TrDeleteMeal.java:53]: exception not explained

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/meals/TrNewPetMeal.java [hunk 16]
```diff
@@ -62,8 +60,7 @@ public boolean isResult() {
     /**
      * Execute the transaction.
      */
-    public void execute() throws MealAlreadyExistingException, InterruptedException, ExecutionException,
-        InvalidFormatException {
+    public void execute() throws MealAlreadyExistingException, InvalidFormatException {
```
Comment (marclll) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/meals/TrNewPetMeal.java:65]: exception not explained

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrAddNewPetIllness.java [hunk 17]
```diff
@@ -63,8 +60,7 @@ public boolean isResult() {
     /**
      * Executes the transaction.
      */
-    public void execute() throws NotPetOwnerException, IllnessAlreadyExistingException, ExecutionException,
-        InterruptedException, InvalidFormatException {
+    public void execute() throws NotPetOwnerException, IllnessAlreadyExistingException {
```
Comment (marclll) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrAddNewPetIllness.java:65]: exception not explained

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrAddNewPetVaccination.java [hunk 18]
```diff
@@ -60,8 +57,7 @@ public boolean isResult() {
         return result;
     }
 
-    public void execute() throws NotPetOwnerException, VaccinationAlreadyExistingException, ExecutionException,
-        InterruptedException, InvalidFormatException {
+    public void execute() throws NotPetOwnerException, VaccinationAlreadyExistingException {
```
Comment (marclll) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrAddNewPetVaccination.java:65]: function not explained

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrDeletePetIllness.java [hunk 19]
```diff
@@ -61,7 +59,7 @@ public boolean isResult() {
     /**
      * Executes the transaction.
      */
-    public void execute() throws NotPetOwnerException, ExecutionException, InterruptedException {
+    public void execute() throws NotPetOwnerException {
```
Comment (marclll) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrDeletePetIllness.java:63]: exception not explained

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrDeletePetVaccination.java [hunk 20]
```diff
@@ -61,7 +59,7 @@ public boolean isResult() {
     /**
      * Executes the transaction.
      */
-    public void execute() throws NotPetOwnerException, ExecutionException, InterruptedException {
+    public void execute() throws NotPetOwnerException {
```
Comment (marclll) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrDeletePetVaccination.java:63]: exception not explained

**Review by marclll — CHANGES_REQUESTED** [review]

**Review by xavier-campos-diaz — APPROVED** [review]

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/user/TrExistsUsername.java  -1
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrObtainAllPetWashes.java  -1
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrUpdateWash.java  -2
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/services/googlecalendar/GoogleCalendarAdapter.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/medicalprofile/MedicalProfileManagerAdapter.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/services/pet/PetManagerAdapter.java  1
         

Clones removed
==============
+ app/src/main/java/org/pesmypetcare/mypetcare/services/vetvisits/VetVisitsAdapter.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5635133&bid=18454556)

**Review by marclll — APPROVED** [review]

**Pull request merged**